pkg/decoders/base64.go

```diff
@@ -24,11 +26,19 @@ func init() {
 }
 
 func (d *Base64) FromChunk(chunk *sources.Chunk) *sources.Chunk {
-	encodedSubstrings := getSubstringsOfCharacterSet(chunk.Data, 20)
+	encodedSubstrings := getSubstringsOfCharacterSet(chunk.Data, 20, b64CharsetMapping, b64EndChars)
 	decodedSubstrings := make(map[string][]byte)
 
 	for _, str := range encodedSubstrings {
 		dec, err := base64.StdEncoding.DecodeString(str)
+		if err == nil {
+			if len(dec) > 0 {
```

Comment:
this is pretty nitty, but breaking this conjunction into two separate `if` statements caused me to take a second to realize that this is the same logic as is being run on line 42. is it possible to write the two checks the same way?

Reply:
```go
	for _, str := range encodedSubstrings {
		dec, err := base64.StdEncoding.DecodeString(str)
		if err == nil {
			if len(dec) > 0 {
				decodedSubstrings[str] = dec
			}
			continue
		}

		dec, err = base64.RawURLEncoding.DecodeString(str)
		if err == nil && len(dec) > 0 {
			decodedSubstrings[str] = dec
		}
	}
```

In context - the first check is intentionally different, because if we successfully decoded with `base64.StdEncoding`, then regardless of whether the string length was 0 or not, we want to `continue` instead of also trying to decode with `base64.RawURLEncoding`. This saves the extra effort of attempting to do the url-safe decoding if we already successfully decoded.

To make these checks the _same_ is possible but either results in:
1. not continuing on the first check, resulting in extra work
2. OR making the second check have an extra nested if block:
```go
		dec, err = base64.RawURLEncoding.DecodeString(str)
		if err == nil {
			if len(dec) > 0 {
				decodedSubstrings[str] = dec
			}
		}
```
But I think both ways of doing this are not ideal.

Do you think we can keep as is or suggest an alternative?

Reply:
I'm sorry, you're right, I just misread. Thanks for the clarification!